context/app/static/js/components/publications/PublicationVignette/PublicationVignette.jsx

```diff
@@ -33,22 +34,46 @@ async function fetchVitessceConf({ assetsEndpoint, uuid, filePath, groupsToken,
   return fillUrls(conf, urlHandler, requestInitHandler);
 }
 
-function PublicationVignette({ vignette, vignetteDirName, uuid }) {
+function PublicationVignette({ vignette, vignetteDirName, uuid, mounted }) {
   const { assetsEndpoint, groupsToken } = useContext(AppContext);
 
   const [vitessceConfs, setVitessceConfs] = useState(undefined);
 
+  // Workaround to make the visualization render only after the accordion section has been expanded while
+  // still letting the prerequisites for the visualizations prefetch
+  const [hasBeenMounted, setHasBeenMounted] = useState(false);
   useEffect(() => {
+    if (mounted) {
+      setHasBeenMounted(true);
+    }
+  }, [mounted]);
+
+  useEffect(() => {
```

Comment:
Also, I think I like the pattern of moving all hooks into a single hook per component. How do you feel about this?

Reply:
That actually sounds great to me, as long as we keep the hooks logically organized and focused on one purpose per hook. e.g, in this case, I'll turn the logic in this component into a `useVitessceConf` hook and a `useHasBeenMounted` hook, as opposed to trying to squeeze both sets of logic into one hook.